File: 0338-counting-bits/0338-counting-bits.java
class Solution {
    public int[] countBits(int n) {
        int[] result = new int[n + 1];
        result[0] = 0;
        for (int i = 1; i <= n ; i++) {
            result[i] = hammingWeight(i);
        } 
        return result;
    }
        
    private int hammingWeight(int n) {
        int ans = 0;
        while (n > 0) {
            if (n % 2 == 1) ans++;
            n /= 2;
        }
        return ans;
    }
}

File: 0338-counting-bits/0338-counting-bits.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        result = [];
        result.append(0)
        i = 1
        while i <= n:
            result.append(self.hammingWeight(i))
            i += 1
        return result
        
        
    def hammingWeight(self, n: int) -> int:
        ans = 0
        while n:
            n &= (n-1)
            ans += 1
        return ans
        